doc/user/bgp.rst

```diff
@@ -3106,6 +3106,48 @@ starting the daemon and the configuration gets saved, the option will persist
 unless removed from the configuration with the negating command prior to the
 configuration write operation.
 
+.. _bgp-suppress-fib:
+
+Suppressing routes not installed in FIB
+=======================================
+
+The FRR implementation of BGP advertises prefixes learnt from a peer to other
+peers even if the routes do not get installed in the FIB. There can be
+scenarios where the hardware tables in some of the routers (along the path from
+the source to destination) is full which will result in all routes not getting
+installed in the FIB. If these routes are advertised to the downstream routers
+then traffic will start flowing and will be dropped at the intermediate router.
+
+The solution is to provide a configurable option to check for the FIB install
+status of the prefixes and advertise to peers if the prefixes are successfully
+installed in the FIB. The advertisement of the prefixes are suppressed if it is
+not installed in FIB.
+
+The following conditions apply will apply when checking for route installation
+status in FIB :
+1. The advertisement or suppression of routes based on FIB install status
+   applies only for newly learnt routes from peer (routes which are not in
+   BGP local RIB).
+2. If the route received from peer already exists in BGP local RIB and route
+   attributes have changed (best path changed), the old path is deleted and
+   new path is installed in FIB. The FIB install status will not have any
+   effect. Therefore only when the route is received first time the checks
+   apply.
+3. The feature will not apply for routes learnt through other means like
+   redistribution to bgp from other protocols. This is applicable only to
+   peer learnt routes.
+4. If a route is installed in FIB and then gets deleted from the dataplane,
+   then routes will not be withdrawn from peers. This will be considered as
+   dataplane issue.
```

Comment:
I'd like some clarity here as this is somewhat confusing to me. Won't Zebra notify BGP about the route deletion and cause BGP to withdraw the route in any case?

Reply:
When the route which is already installed in dataplane gets deleted, zebra gets the message. Currently we are not handling this in the PR as it is less likely that route which is successfully installed in dataplane to get deleted for some reason.
We can handle it this in seperate PR later if required.